### PR TITLE
Add Support for BPF_PROG_TYPE_CGROUP_SOCK_ADDR

### PR DESCRIPTION
**Pull request by dave-tucker**
Sample program: https://github.com/dave-tucker/cgroup-sock-test

Closes: #213

### CODE REVIEW

**Review by alessandrod — CHANGES_REQUESTED** [review]
Great stuff! See comments.

Also meta comment: I think this should have `Cgroup` and `cgroup_` prefixes like the other `CGROUP_` programs?

File: aya/src/obj/mod.rs [hunk 1]
```diff
@@ -104,30 +105,82 @@ pub(crate) struct Function {
...
-    CgroupSysctl { name: String },
-    LircMode2 { name: String },
-    PerfEvent { name: String },
-    RawTracePoint { name: String },
-    Lsm { name: String },
-    BtfTracePoint { name: String },
-    FEntry { name: String },
-    FExit { name: String },
-    Extension { name: String },
+    KRetProbe {
```
[the remark this answers is not in the document]
Comment (dave-tucker) [reply aya/src/obj/mod.rs:108]: It was rustfmt 😅
Let me see if I can convince it not to do this

[the remark this answers is not in the document]
Comment (alessandrod) [reply aya/src/obj/mod.rs:108]: Nah it's fine! As long as it does it for everyone and we don't flip flop it at every commit

File: aya/src/programs/sock_addr.rs [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+            "connect6" => Ok(SockAddrAttachType::Connect6),
+            "getpeername4" => Ok(SockAddrAttachType::GetPeerName4),
+            "getpeername6" => Ok(SockAddrAttachType::GetPeerName6),
+            "getsockname4" => Ok(SockAddrAttachType::GetSockName4),
+            "getsockname6" => Ok(SockAddrAttachType::GetSockName6),
+            "sendmsg4" => Ok(SockAddrAttachType::UDPSendMsg4),
+            "sendmsg6" => Ok(SockAddrAttachType::UDPSendMsg6),
+            "recvmsg4" => Ok(SockAddrAttachType::UDPRecvMsg4),
+            "recvmsg6" => Ok(SockAddrAttachType::UDPRecvMsg6),
+            _ => Err(ProgramError::InvalidAttachType {
```
[the remark this answers is not in the document]
Comment (dave-tucker) [reply aya/src/programs/sock_addr.rs:227]: I've moved it to `ParseError`, which isn't "exactly" what you asked :laughing: but is the right place for this error IMHO. 
clippy complained that the `pub(crate) struct InvalidAttachType` would be leaked so wouldn't allow your suggestion.

**Comment by dave-tucker** [comment]
@alessandrod I've addressed most of your comments if you'd like to take another look.

File: aya/src/programs/sock_addr.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (alessandrod) [reply aya/src/programs/sock_addr.rs]: I'd make it `pub(crate) try_from` at the `impl CgroupSockAddrAttachType` and return a `pub(crate)` error (without actually implementing `TryFrom` for `CgroupSockAddrAttachType`). We are now exposing a public variant for an internally used error, no bueno.

**Pull request merged**